### PR TITLE
Update Developer Guide and User Guide

### PR DESCRIPTION
**Pull request by zj-cs2103**
Add add feature to developer guide and update commands for staff for user guide.
Placeholder for sequence diagram for now, will be added in next milestone.

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -231,6 +231,48 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+**Aspect: Implementation of the four add commands**
+
+* **Alternative 1 (current choice)**: four separate add commands, one for each type of item
+** Pros:
+*** Easier implementation
+** Cons:
+*** Causes code duplication, as all the add commands differ only by their parameters, and function similarly otherwise.
+*** More types of commands for the user to remember
+
+* **Alternative 2**: a single add command, with a type field to indicate the type of item to be added
```
Comment (hiyingnn) [inline docs/DeveloperGuide.adoc:265]: Possibly can talk about how the nature of all items is different, thus requiring different fields, prefixes etc, not suitable for just 1 command.

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -231,6 +231,48 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+The add feature allows the user to add items to the `RestaurantBook`.
+It also populates the item added to the `RestaurantBook` with parameters supplied by the user.
+
+==== Current Implementation
+
+The add feature is implemented as four separate commands, one for adding each type of item: `addMember`, `addBooking`, `addIngredient`, `addStaff`.
+Each command is implemented as a `AddCommand` class. For example, the `addMember` command is implemented as the `AddMember` class.
+All `AddCommand` class inherit from the `Command` class.
+As the general flow of the various add commands are similar, the `addMember` command will be used as an example here.
```
Comment (hiyingnn) [inline docs/DeveloperGuide.adoc:244]: If going into detail, and using sequence diagram to illustrate, maybe can talk about Parser as well

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -231,6 +231,48 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+*** Easier implementation
+** Cons:
+*** Causes code duplication, as all the add commands differ only by their parameters, and function similarly otherwise.
+*** More types of commands for the user to remember
+
+* **Alternative 2**: a single add command, with a type field to indicate the type of item to be added
+** Pros
+*** Reduces the number of distinct commands in the RestaurantBook, thus making the `RestaurantBook` a simpler application for users.
+** Cons
+*** Complicates the add command as the parameters for the add command depend on the command type parameter passed into the command
```
Comment (lingyanhao) [inline docs/DeveloperGuide.adoc:269]: complicates the parser?
the add itself can be done the same way right

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -231,6 +231,48 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+=== Add Feature
+
+The add feature allows the user to add items to the `RestaurantBook`.
+It also populates the item added to the `RestaurantBook` with parameters supplied by the user.
+
+==== Current Implementation
+
+The add feature is implemented as four separate commands, one for adding each type of item: `addMember`, `addBooking`, `addIngredient`, `addStaff`.
+Each command is implemented as a `AddCommand` class. For example, the `addMember` command is implemented as the `AddMember` class.
+All `AddCommand` class inherit from the `Command` class.
```
Comment (lingyanhao) [inline docs/DeveloperGuide.adoc:243]: A little confusing- readers might think that `AddCommand` is an actual class.

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -231,6 +231,48 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+** Cons:
+*** Causes code duplication, as all the add commands differ only by their parameters, and function similarly otherwise.
+*** More types of commands for the user to remember
+
+* **Alternative 2**: a single add command, with a type field to indicate the type of item to be added
+** Pros
+*** Reduces the number of distinct commands in the RestaurantBook, thus making the `RestaurantBook` a simpler application for users.
+** Cons
+*** Complicates the add command as the parameters for the add command depend on the command type parameter passed into the command
+*** May have to display a long usage guide comprising examples of adding all four types of items
```
Comment (lingyanhao) [inline docs/DeveloperGuide.adoc:270]: can also consider talking about context sensitive parsing

**Pull request merged**